j5/backends/console/sr/v4/power_board.py

```diff
@@ -0,0 +1,146 @@
+
+    def get_power_output_enabled(self, identifier: int) -> bool:
+        """Get whether a power output is enabled."""
+        try:
+            return self._output_states[identifier]
+        except KeyError:
+            raise ValueError(f"Invalid power output identifier {identifier!r}; "
+                             f"valid identifiers are "
+                             f"{self._output_states.keys()}") from None
+
+    def set_power_output_enabled(
+        self, identifier: int, enabled: bool,
+    ) -> None:
+        """Set whether a power output is enabled."""
+        try:
+            self._console.info(f"Setting output {identifier} to {enabled}")
+            self._output_states[identifier] = enabled
+        except KeyError:
+            raise ValueError(f"Invalid power output identifier {identifier!r}; "
+                             f"valid identifiers are "
+                             f"{self._output_states.keys()}") from None
+
+    def get_power_output_current(self, identifier: int) -> float:
+        """Get the current being drawn on a power output, in amperes."""
+        if identifier in self._output_states:
+
+            return self._console.read(f"Power Output current for {identifier}")
+        else:
+            raise ValueError(f"Invalid power output identifier {identifier!r}; "
+                             f"valid identifiers are "
+                             f"{self._output_states.keys()}") from None
+
+    def buzz(self, identifier: int,
+             duration: timedelta, pitch: int) -> None:
+        """Queue a pitch to be played."""
+        if identifier != 0:
+            raise ValueError(f"invalid piezo identifier {identifier!r}; "
+                             f"the only valid identifier is 0")
+        duration_ms = round(duration / timedelta(milliseconds=1))
+        if duration_ms > 65535:
+            raise ValueError("Maximum piezo duration is 65535ms.")
+        self._console.info(f"Buzzing at {pitch}Hz for {duration_ms}ms")
+
+    def get_button_state(self, identifier: int) -> bool:
+        """Get the state of a button."""
+        if identifier != 0:
+            raise ValueError(f"invalid button identifier {identifier!r}; "
+                             f"the only valid identifier is 0")
+        return self._console.read("Start button state", bool)
+
+    def wait_until_button_pressed(self, identifier: int) -> None:
+        """Halt the program until this button is pushed."""
+        self._console.info("Waiting for start button press.")
+        while not self.get_button_state(identifier):
+            sleep(0.05)
+
+    def get_battery_sensor_voltage(self, identifier: int) -> float:
+        """Get the voltage of a battery sensor."""
+        if identifier != 0:
+            raise ValueError(f"invalid battery sensor identifier {identifier!r}; "
+                             f"the only valid identifier is 0")
+        return self._console.read("Battery Sensor Voltage", float)
```

Comment:
I think we should make it clear to the user what they're expected to type in (i.e. type or units of value).
```suggestion
        return self._console.read("Battery voltage [volts]", float)
```